// enkanetwork.js/dist/structs/AssetFinder/StarRail/SRCharacter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SRCharacterSkills = exports.SRCharacterImages = exports.SRCharacterAssets = void 0;
const utils_1 = require("../../../utils");
const characters = utils_1.srcharacters;
const skills = utils_1.srskills;
/**
 * A class that structures the character assets and name.
 */
class SRCharacterAssets {
    /**
     * The name of the character.
     */
    name;
    /**
     * The assets of the character.
     */
    assets;
    /**
     * Creates a new `SRCharacterAssets` instance.
     * @param characterId - The ID of the character.
     * @param language - The language to get the name.
     */
    constructor(characterId, language) {
        const character = characters[characterId];
        this.name = utils_1.starrailFinder[language].hash(character?.AvatarName).value;
        this.assets = character
            ? new SRCharacterImages(character, language)
            : {};
    }
}
exports.SRCharacterAssets = SRCharacterAssets;
class SRCharacterImages {
    /**
     * The icon path of the character.
     */
    icon;
    /**
     * The gacha splash icon path of the character.
     */
    gachaIcon;
    /**
     * The paths of all the eidolon icons of the character.
     */
    eidolons;
    /**
     * The paths for the skill icons of the character.
     */
    skills;
    /**
     * The name and paths of the character skins assets.
     */
    skins;
    /**
     * Creates a new `SRCharacterImages` instance.
     * @param data - The data of the character assets.
     */
    constructor(data, language) {
        this.icon = data.AvatarSideIconPath;
        this.gachaIcon = data.AvatarCutinFrontImgPath;
        this.eidolons = data.RankIDList.map((id) => utils_1.starrailFinder.en.eidolon(id).icon);
        this.skills = new SRCharacterSkills(data.SkillList);
        this.skins = data.Skins
            ? data.Skins.map((data) => new SRCharacterSkins(data, language))
            : [];
    }
}
exports.SRCharacterImages = SRCharacterImages;
class SRCharacterSkills {
    /**
     * The path of the basic attack icon.
     */
    basicAttack;
    /**
     * The path of the skill icon.
     */
    skill;
    /**
     * The path of the ultimate icon.
     */
    ultimate;
    /**
     * The path of the talent icon.
     */
    talent;
    /**
     * The path of the technique icon.
     */
    technique;
    /**
     * Creates a new `SRCharacterSkills` instance.
     * @param data - The data of the character skills.
     */
    constructor(data) {
        this.basicAttack = skills[data[0]].SkillIcon;
        this.skill = skills[data[1]].SkillIcon;
        this.ultimate = skills[data[2]].SkillIcon;
        this.talent = skills[data[3]].SkillIcon;
        this.technique = skills[data[5]].SkillIcon;
    }
}
exports.SRCharacterSkills = SRCharacterSkills;
class SRCharacterSkins {
    /**
     * The ID of the skin.
     */
    id;
    /**
     * The name of the skin.
     */
    name;
    /**
     * The path to the full image of the skin.
     */
    fullImage;
    /**
     * The path to the icon of the skin.
     */
    icon;
    constructor(data, language) {
        this.id = data.ID;
        this.name = utils_1.starrailFinder[language].hash(data.AvatarSkinName).value;
        this.fullImage = data.AvatarCutinFrontImgPath;
        this.icon = data.AvatarSideIconPath;
    }
}
